### PR TITLE
fix: Empty swap table with valid swaps

### LINKED ISSUE #11296: Empty swap table with valid swaps
### How to reproduce 

- create swap
- wait for the page redirect to the profile page
- wait for the indexer to be updated,  if you have no swaps  the tab will update but the rows will not

CleanShot 2024-12-25 at 16 23 31@2x

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [x] Bugfix
- [ ] Feature
- [x] Refactoring

## Context

Adds the ability to pass `variables` as `ref` to `useGraphql`

reviewers kindly also test , tyvm

- removed: unsued code `useNewCollectionId ` 
- fix: gallery item refetch with wrong params
- [x] Closes #11296

## Screenshot 📸

- [x] My fix has changed **something** on UI; 

CleanShot 2024-12-26 at 13 23 40

### CODE REVIEW

**Comment by hassnian** [comment]
@preschian any idea why the preview link is not working? 

CleanShot 2024-12-28 at 10 44 40@2x

**Comment by exezbcz** [comment]
is not showing the skeleton first better and then fallback to the empty message?

**Comment by preschian** [comment]
> @preschian any idea why the preview link is not working?

hhmm, this is new for me. seems like need to check it through netlify dashboard cc @vikiival

File: composables/useGraphql.ts [hunk 1]
```diff
@@ -1,56 +1,49 @@
...
     try {
       loading.value = true
-      const { data: result } = await useAsyncQuery<T>({
-        query: query.default,
-        variables: {
-          ...variables,
-          ...extraVariables,
-        },
-        clientId: isRef(client) ? String(client.value) : client,
+      const requestId = ++currentRequestId.value
```
Comment (preschian) [inline composables/useGraphql.ts:33]: We probably don't need the `useGraphql` composable at all. I prefer using `useQuery` directly from NuxtApollo instead - that composable alone might be sufficient

File: composables/transaction/mintCollection/useNewCollectionId.ts [hunk 2]
```diff
@@ -1,36 +1,3 @@
-import { getRandomValues } from '@/components/unique/utils'
-import { mapToId } from '@/utils/mappers'
-import { unwrapSafe } from '@/utils/uniquery'
-
-export function useNewCollectionId() {
```
Comment (preschian) [inline composables/transaction/mintCollection/useNewCollectionId.ts:5]: nit: let's move to separate PR regarding remove unused stuff

**Review by preschian — COMMENTED** [review]
Anyway, are you able to create a branch directly in kodadot/nft-gallery instead of branching from a fork?

This would allow us to see the deployment preview through Cloudflare Pages, since currently we're not able to see the deployment preview from Netlify. At the moment, Cloudflare Pages doesn't support deployment previews from forked branches

**Comment by hassnian** [comment]
closing in favour of  #11303

File: composables/useGraphql.ts (hunk 1 above)
Comment (hassnian) [reply composables/useGraphql.ts:33]: Yep, I like this solution a lot better.

File: composables/transaction/mintCollection/useNewCollectionId.ts (hunk 2 above)
Comment (hassnian) [reply composables/transaction/mintCollection/useNewCollectionId.ts:5]: sure
